### PR TITLE
Add more code quality checks

### PR DESCRIPTION
**Pull request by sfc-gh-turbaszek**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Bump ruff version, add more check in pre-commit

### CODE REVIEW

**Review by sfc-gh-pjob — CHANGES_REQUESTED** [review]

File: src/snowcli/cli/containers/jobs/commands.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def create(
 @app.command()
 @global_options_with_connection
 def logs(
-    id: str = typer.Argument(..., help="Job id"),
+    idx: str = typer.Argument(..., help="Job id"),
```
[the remark this answers is not in the document]
Comment (sfc-gh-turbaszek) [reply src/snowcli/cli/containers/jobs/commands.py:44]: Very good comment, fixed

File: tests/testing_utils/fixtures.py [hunk 2]
```diff
@@ -32,9 +32,9 @@ def __init__(self, app: Typer, test_snowcli_config: str):
 
     @functools.wraps(CliRunner.invoke)
     def invoke(self, *a, **kw):
-        return self.invoke_with_config_file(self.test_snowcli_config, *a, **kw)
+        return self.invoke_file(self.test_snowcli_config, *a, **kw)
 
-    def invoke_with_config_file(self, config_file: str, *a, **kw):
+    def invoke_file(self, config_file: str, *a, **kw):
```
[the remark this answers is not in the document]
Comment (sfc-gh-turbaszek) [reply tests/testing_utils/fixtures.py:37]: I think this is a bad refactor, fixing too many fixtures of the same name and duplicated code is also on my todo list 😄

**Review by sfc-gh-pjob — DISMISSED** [review]

File: pyproject.toml [hunk 3]
```diff
@@ -71,7 +71,14 @@ packages = ["src/snowcli", "src/templates"]
 line-length = 88
 select = [
     "N",
-    "I"  # isort
+    "I",  # isort
+    "G",  # flake8-logging-format
```
Comment (sfc-gh-cgorrie) [inline pyproject.toml:75]: Reading up on this, [the spec](https://pypi.org/project/flake8-logging-format/) says that part of the motivation is to make sure we can redact logs. Do we have the other pieces in place to make this happen? Is that our motivation for adopting this too? I am a big f-string fan, but if they're not doing what we need them to do, I agree with the change. Unfortunately without this context the change to using e.g. `%s` feels like a step backwards in code readability.

**Review by sfc-gh-cgorrie — COMMENTED** [review]
Some thoughts -- thanks for enforcing more code quality at the tooling level 💪

File: pyproject.toml (hunk 3 above)
Comment (sfc-gh-turbaszek) [reply pyproject.toml:75]: This applies only to logging. Using `%s` is a common practices in python as rendering the string is defferred. Here are docs: https://docs.python.org/3/howto/logging.html#optimization

File: src/snowcli/cli/object/stage/manager.py [hunk 4]
```diff
@@ -73,7 +73,7 @@ def put(
             f"auto_compress=false parallel={parallel} overwrite={overwrite}"
         )
 
-    def _put(
+    def put_with_role(
```
[the remark this answers is not in the document]
Comment (sfc-gh-turbaszek) [reply src/snowcli/cli/object/stage/manager.py:76]: We can either make the role mandatory (pushing implementation details of if role then to other place) or we can add this functionality to `put` method. WDYT?

File: pyproject.toml (hunk 3 above)
Comment (sfc-gh-cgorrie) [reply pyproject.toml:75]: Sounds good -- so for us the main argument is optimization. I suppose getting rid of f-strings even when not necessary (e.g. for things that aren't heavy to compute) is just an unfortunate side-effect of the linter's abilities.

File: src/snowcli/cli/object/stage/manager.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (sfc-gh-cgorrie) [reply src/snowcli/cli/object/stage/manager.py:76]: For me, I see the easiest fix as just naming it `put`. I don't think we'll always want to change our role for it; we have other primitives to do this (e.g. `SqlExecutionMixin.use_role`).

File: pyproject.toml (hunk 3 above)
Comment (sfc-gh-turbaszek) [reply pyproject.toml:75]: Apart from optimisation you can create a custom formatter that for example highlights "input" to log message. This wouldn't be possible with f-strings

**Review by sfc-gh-astus — APPROVED** [review]

**Review by sfc-gh-astus — DISMISSED** [review]

**Review by sfc-gh-astus — APPROVED** [review]

**Review by sfc-gh-cgorrie — APPROVED** [review]
LGTM

**Review by sfc-gh-bgoel — APPROVED** [review]
LGTM

File: src/snowcli/utils/threadsafe.py [hunk 5]
```diff
@@ -11,7 +11,7 @@ def __init__(self, value: T):
         self._value = value
         self._lock = threading.Lock()
 
-    def set(self, new_value: T) -> T:
+    def set(self, new_value: T) -> T:  # noqa: A003
```
Comment (sfc-gh-mraba) [inline src/snowcli/utils/threadsafe.py:14]: I'm wondering what is the source of this `A003` warning?
Neither flake8, pylint nor mypy reports this issue.

```
snowcli on  SNOW-986249-option-for-intermediate-output-of-command [$!] via 🐍 v3.8.18 (py3.8) took 13s
❯ flake8 src/snowcli/utils/threadsafe.py

snowcli on  SNOW-986249-option-for-intermediate-output-of-command [$!] via 🐍 v3.8.18 (py3.8)
❯ mypy !$
mypy src/snowcli/utils/threadsafe.py
Success: no issues found in 1 source file

snowcli on  SNOW-986249-option-for-intermediate-output-of-command [$!] via 🐍 v3.8.18 (py3.8)
❯ pylint !$
pylint src/snowcli/utils/threadsafe.py
************* Module src.snowcli.utils.threadsafe
```

_pylint only complains about docstrings_

**Review by sfc-gh-mraba — APPROVED** [review]

File: src/snowcli/utils/threadsafe.py (hunk 5 above)
Comment (sfc-gh-turbaszek) [reply src/snowcli/utils/threadsafe.py:14]: https://pypi.org/project/flake8-builtins/
```
A003 Class attribute `set` is shadowing a Python builtin
```

**Pull request merged**